Sources/WhoopDIKit/Container/Container.swift

```diff
@@ -1,23 +1,49 @@
 import Foundation
 
 public final class Container {
-    private let localDependencyGraph: ThreadSafeDependencyGraph
-    private var isLocalInjectActive: Bool = false
     private let options: WhoopDIOptionProvider
-    
+    private let parent: Container?
+
     private let serviceDict = ServiceDictionary<DependencyDefinition>()
 
     /// Creates a container and registers a list of modules with the DI system.
     /// Typically you will create a `DependencyModule` for your feature, then add it to the module list provided to this method.
     /// Each provided module and it's dependencies will be registered with the DI system.
     /// Dependencies are registered in dependency order, with leaf modules (those with no dependencies) being registered first.
     public init(modules: [DependencyModule] = [],
+                parent: Container? = nil,
                 options: WhoopDIOptionProvider = defaultWhoopDIOptions()) {
+        self.parent = parent
         self.options = options
-        localDependencyGraph = ThreadSafeDependencyGraph(options: options)
         registerModules(modules: modules)
     }
 
+    /// Creates a container with a local module definition closure.
+    /// This initializer allows you to define dependencies inline without creating a separate module.
+    /// The dependencies defined in the closure will be registered with the DI system.
+    ///
+    /// Example:
+    /// ```swift
+    /// let container = Container { module in
+    ///     module.factory { MyService() }
+    ///     module.singleton { SharedService() }
+    /// }
+    /// ```
+    ///
+    /// - Parameters:
+    ///   - parent: An optional parent container to inherit dependencies from
+    ///   - options: Configuration options for the DI system
+    ///   - localDefinition: A closure that defines the dependencies for this container
+    public init(parent: Container? = nil,
+                options: WhoopDIOptionProvider = defaultWhoopDIOptions(),
+                _ localDefinition: (DependencyModule) -> Void) {
+        self.parent = parent
```

Comment:
are we using the parent anywhere? I might be missing it

Reply:
Yup - within get (on line 106). We first get from the current dependency graph, then the parent if it's not there.